OtripleS.Web.Api.Tests.Acceptance/APIs/TeacherAttachments/TeacherAttachmentsApiTests.Logic.cs

```diff
@@ -0,0 +1,95 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using OtripleS.Web.Api.Tests.Acceptance.Models.TeacherAttachments;
+using RESTFulSense.Exceptions;
+using Xunit;
+
+namespace OtripleS.Web.Api.Tests.Acceptance.APIs.TeacherAttachments
+{
+    public partial class TeacherAttachmentsApiTests
+    {
+        [Fact]
+        public async Task ShouldPostTeacherAttachmentAsync()
+        {
+            // given
+            TeacherAttachment randomTeacherAttachment = await CreateRandomTeacherAttachment();
+            TeacherAttachment inputTeacherAttachment = randomTeacherAttachment;
+            TeacherAttachment expectedTeacherAttachment = inputTeacherAttachment;
+
+            // when             
+            await this.otripleSApiBroker.PostTeacherAttachmentAsync(inputTeacherAttachment);
+
+            TeacherAttachment actualTeacherAttachment =
+                await this.otripleSApiBroker.GetTeacherAttachmentByIdsAsync(
+                    inputTeacherAttachment.TeacherId,
+                    inputTeacherAttachment.AttachmentId);
```

Comment:
We need to verify both.
The object that is returned from the API call.
And the object that actually persisted in the database.

Here's a good example:

```csharp
[Fact]
        public async Task ShouldPostPhoneAsync()
        {
            // given
            Phone randomPhone = await CreateRandomPhoneAsync();
            Phone inputPhone = randomPhone;
            Guid inputProfileId = inputPhone.CreatedBy;
            Phone expectedPhone = inputPhone;

            // when
            Phone actualPhone =
                await this.estimizorApiBroker.PostPhoneAsync(inputProfileId, inputPhone);

            Phone retrievedPhone =
                await this.estimizorApiBroker.GetPhoneByIdAsync(inputPhone.Id);

            // then
            actualPhone.Should().BeEquivalentTo(expectedPhone);
            retrievedPhone.Should().BeEquivalentTo(expectedPhone);
            await DeletePhoneAsync(actualPhone);
        }

```